sklearn/ensemble/_bagging.py

```diff
@@ -199,6 +199,11 @@ def _parallel_predict_regression(estimators, estimators_features, X):
     )
 
 
+def _estimator_has(attr):
+    """Check if self.base_estimator has attr"""
+    return lambda self: hasattr(self.base_estimator, attr)
```

Comment:
In general we have to check one of the fitted estimators if available, and back off to the unfitted base estimator.

Reply:
This makes sense. In this particular case (`bagging`), I think the priority may actually be `estimators_` > `base_estimator_` > `base_estimator`, given `base_estimator_` (default if `base_estimator=None`) is set in a lazy fashion. What do you think?

Reply:
Hmm, applying a similar change to stacking (i.e. check fitted `final_estimator_` first and then back off to unfitted `final_estimator`) breaks this [test](https://github.com/scikit-learn/scikit-learn/blob/3ae7c7615343bbd36acece57825d8b0d70fd9da4/sklearn/semi_supervised/tests/test_self_training.py#L319-L337), which it assumes the estimator does not expose `predict_proba` (or `predict`) before it is fitted. Should we modify the test or ONLY check fitted `final_estimator_` like before?

Reply:
I think the test is flawed. There's no reason in that context that StackingClassifier shouldn't be able to work out that it supports predict_proba. The classic (although obscure) case where you can't know is `GridSearchCV(SGDClassifier(), {"loss": ["hinge", "log"]})`.

Reply:
Gotcha. That is a good example.